HotelApi/Controllers/ReservationController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Http;
+using System.Web.Http.Results;
+using HotelBooking;
+using HotelBooking.BookingExternalService;
+using HotelBooking.HotelFactory;
+using HotelBooking.HotelManager;
+using HotelBooking.Logger;
+using HotelBooking.PaymentExternalService;
+
+namespace HotelApi.Controllers
+{
+    public class ReservationController : ApiController
+    {
+        private readonly HotelManager _manager;
+
+        public ReservationController()
+        {
+            IHotelFactory factory = new HotelFactory();
+            IPaymentService paymentService = new PaymentService();
+            IBookingService bookingService = new BookingService();
+            ILogger logger = new ConsoleLogger();
+            _manager = new HotelManager(factory, paymentService, bookingService, logger);
+        }
+
+        [HttpGet]
```

Comment:
Please note:

If you will follow convention you do not need to use HttpGet attribute

Reply:
applied